### DISCUSSION TITLE
Is it possible to create field arrays with nested objects?

### DISCUSSION THREAD

**Discussion by samuelthng** [react-hook-form/react-hook-form#2553] [Q&A]
Currently, I'm having a use case where I have nested useFieldArrays, in which the inner field array requires an object with the following shape:

{
  modeA: { ...modeAvalues},
  modeB: { ...modeBvalues},
  ...
}

The issue is that, since we are required to set the default value as field[inputName] and if the input name is `modeA.example`, the component throws a not a property of undefined error.

Has anyone faced this or see an alternative solution besides mapping the values outside the form?

**Comment by bluebill1049** [comment]
do you have a codesandbox for this issue?

**Reply by samuelthng** [reply]
I found my mistake when making the sandbox. So when I append an entry into the fieldArray, the appended object has to conform to the shape of the form I'm using or it will break.
